### PR TITLE
add distributed job api

### PR DESCRIPTION
**Pull request by tornado404**

### CODE REVIEW

File: pkg/common/schema/job.go [hunk 1]
```diff
@@ -286,6 +288,10 @@ func (c *Conf) Type() JobType {
 	return JobType(c.Env[EnvJobType])
 }
 
+func (c *Conf) Framework() Framework {
```
Comment (D0m021ng) [inline pkg/common/schema/job.go]: Add `preCheck` for env to avoid panic.

File: pkg/common/schema/job.go [hunk 2]
```diff
@@ -181,6 +182,7 @@ type PFJobConf interface {
 	SetAnnotations(string, string)
 
 	Type() JobType
+	Framework() Framework
```
Comment (D0m021ng) [inline pkg/common/schema/job.go]: Why need `Framework`?

Comment (tornado404) [reply pkg/common/schema/job.go]: removed

File: pkg/apiserver/controller/job/job.go [hunk 3]
```diff
@@ -164,6 +164,7 @@ func patchEnvs(conf *schema.Conf, commonJobInfo *CommonJobInfo) error {
 		return err
 	}
 	conf.SetQueueID(queue.ID)
+	conf.SetPriority(commonJobInfo.SchedulingPolicy.Priority)
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/job.go]: Repeat with line `169-171`

File: pkg/apiserver/controller/job/job.go [hunk 4]
```diff
@@ -164,6 +164,7 @@ func patchEnvs(conf *schema.Conf, commonJobInfo *CommonJobInfo) error {
...
 	conf.SetEnv(schema.EnvJobQueueName, queue.Name)
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/job.go]: Repeat with QueueID

File: pkg/apiserver/controller/job/job.go [hunk 5]
```diff
@@ -177,6 +178,7 @@ func patchEnvs(conf *schema.Conf, commonJobInfo *CommonJobInfo) error {
 func patchSingleEnvs(conf *schema.Conf, request *CreateSingleJobRequest) error {
 	log.Debugf("patchSingleEnvs conf=%#v, request=%#v", conf, request)
 	// fields in request.CommonJobInfo
+	conf.SetEnv(schema.EnvJobType, string(schema.TypeSingle))
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/job.go]: Using models.Job filed `Type`

File: pkg/apiserver/controller/job/job.go [hunk 6]
```diff
@@ -197,16 +199,195 @@ func patchSingleEnvs(conf *schema.Conf, request *CreateSingleJobRequest) error {
 	}
 	conf.SetFlavour(flavour.Name)
```
Comment (D0m021ng) [inline pkg/apiserver/controller/job/job.go]: Using conf.Flavour

Comment (tornado404) [reply pkg/apiserver/controller/job/job.go]: done

File: pkg/common/schema/job.go [hunk 7]
```diff
@@ -25,6 +25,7 @@ type MemberRole string
 
 const (
 	EnvJobType        = "PF_JOB_TYPE"
+	EnvJobFramework   = "PF_JOB_FRAMEWORK"
```
Comment (D0m021ng) [inline pkg/common/schema/job.go]: Remove it

**Review by D0m021ng — APPROVED** [review]

**Pull request merged**